src/fondant/pipeline/runner.py

```diff
@@ -66,22 +67,62 @@ def run(
             to mount in the docker-compose spec.
             build_args: List of build arguments to pass to docker
         """
-        if isinstance(input, Pipeline):
-            os.makedirs(".fondant", exist_ok=True)
-            output_path = ".fondant/compose.yaml"
-            logging.info(
-                "Found reference to un-compiled pipeline... compiling",
+        if self.docker_is_installed() and self.docker_compose_is_installed():
+            if isinstance(input, Pipeline):
+                os.makedirs(".fondant", exist_ok=True)
+                output_path = ".fondant/compose.yaml"
+                logging.info(
+                    "Found reference to un-compiled pipeline... compiling",
+                )
+                compiler = DockerCompiler()
+                compiler.compile(
+                    input,
+                    output_path=output_path,
+                    extra_volumes=extra_volumes,
+                    build_args=build_args,
+                )
+                self._run(output_path)
+            else:
+                self._run(input)
+
+    @staticmethod
+    def docker_is_installed():
+        """Execute docker command to check if docker is available."""
+        try:
+            # Check Docker info
+            subprocess.check_call(  # nosec
+                ["docker", "info"],
+                stdout=subprocess.DEVNULL,
             )
-            compiler = DockerCompiler()
-            compiler.compile(
-                input,
-                output_path=output_path,
-                extra_volumes=extra_volumes,
-                build_args=build_args,
+
+        except subprocess.CalledProcessError:
+            sys.exit(
```

Comment:
Is there a reason to use `sys.exit()` and not raise the error ?

Reply:
Haha, I proposed this in a previous review. I think it's more user friendly to show a single error message instead of a full stack trace. At least for the CLI. For the SDK I could go either way.

If we want to have a stacktrace in the SDK, we could raise the error and catch it in the CLI instead.